middleware/security_context_setter.rb

```diff
@@ -1,16 +1,29 @@
 module CloudFoundry
   module Middleware
     class SecurityContextSetter
+      PACKAGES_UPLOAD_REGEX = %r{/v3/packages/[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}/upload}i.freeze
+
       def initialize(app, security_context_configurer)
         @app                         = app
         @security_context_configurer = security_context_configurer
       end
 
       def call(env)
         header_token = env['HTTP_AUTHORIZATION']
+        request_path = env['REQUEST_PATH']
 
         @security_context_configurer.configure(header_token)
 
+        if request_path && request_path.match(PACKAGES_UPLOAD_REGEX)
+          upload_start_time = Rack::Request.new(env).params['upload_start_time'].to_i
+          if upload_start_time
+            auth                  = env['HTTP_AUTHORIZATION']
```

Comment:
The `HTTP_AUTHORIZATION` env is already available in variable `header_token`.

Reply:
Good point!